api/src/OpenTelemetry/Internal/Trace/Types.hs

```diff
@@ -34,23 +34,18 @@ import OpenTelemetry.Trace.TraceState
 import OpenTelemetry.Util
 
 
-data ExportResult
-  = Success
-  | Failure (Maybe SomeException)
-
-
-data Exporter a = Exporter
-  { exporterExport :: HashMap InstrumentationLibrary (Vector a) -> IO ExportResult
-  , exporterShutdown :: IO ()
+data SpanExporter = SpanExporter
+  { spanExporterExport :: HashMap InstrumentationLibrary (Vector ImmutableSpan) -> IO ExportResult
+  , spanExporterShutdown :: IO ()
   }
```

Comment:
Notice the loss of the type variable. `Exporter ImmutableSpan` is equivalent to `SpanExporter`. `Exporter a` never had a different value than `Exporter ImmutableSpan`.